app/src/main/java/com/example/musicapp/screen/MainActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.example.musicapp.screen
+
+import android.Manifest
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.IBinder
+import com.example.musicapp.dataSource.model.Song
+import com.example.musicapp.databinding.ActivityMainBinding
+import com.example.musicapp.screen.listSong.SongListFragment
+import com.example.musicapp.screen.play.PlayFragment
+import com.example.musicapp.service.MusicService
+
+class MainActivity : AppCompatActivity(), SongListFragment.OnDataPass, PlayFragment.OnDataPass {
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var songList: MutableList<Song>
+    private lateinit var callBack: (Int, Song, Boolean) -> Unit
+    lateinit var musicService: MusicService
+    var serviceBound = false
```

Comment:
biến này để public làm gì đấy 
```suggestion
    var isServiceBound = false
```

Reply:
Để em check service đã kết nối chưa. Em log thì thấy thường là fragment gọi đến service bị null rồi 1s sau cái log trong onServiceConnected mới in ra